feat: Add Message Feedback Configuration and Logic

timmanik opened the pull request:
## Summary

Currently, there is not a way to disable user feedback on responses (thumbs up/down) without making direct changes to the source code. Though some admins want to collect this type of user feedback, there are some admins that don't. This PR allows admins to have the option to either enable user feedback or to disable it entirely. 

## Change Type


- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Testing

I introduced a new configuration option in the interface settings that can control message feedback visibility. This can be configured in through `librechat.yaml` configuration:
```yaml
interface:
  messageFeedback: false
```
When this setting is `false`, the user feedback buttons (thumbs up/down) will be hidden from the chat UI. When this setting is either `true` or not set at all (default behavior), the feedback buttons will be visible for AI responses.

Screenshot below is when **messageFeedback: false** in `librechat.yaml`.

<img width="808" alt="image" src="https://github.com/user-attachments/assets/5ccb697a-9c68-40d6-b9ad-06d6f073ceaf" />

### **Test Configuration**:

1. Set `messageFeedback: false` in your `librechat.yaml` or set `INTERFACE_MESSAGE_FEEDBACK=false` in your `.env`
2. Start the application
3. Verify that no feedback buttons (thumbs up/down) appear on any AI messages
4. Set the value back to `true` or remove the configuration
5. Verify that feedback buttons reappear on AI messages

## Checklist

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [ ] I have made pertinent documentation changes
- [x] My changes do not introduce new warnings
- [x] Local unit tests pass with my changes
- [ ] A pull request for updating the documentation has been submitted.

rubentalstra commented:
@timmanik hi, thank you for your contribution. can you please add this in the `YAML` file instead of the `.ENV` so we can intergate this later in the admin panel easier 😄

timmanik committed:
feat: Add message feedback configuration to interface and components

- Added `messageFeedback` option to the interface configuration in `librechat.example.yaml` and `config.ts`.
- Updated `AppService.spec.js` to include `messageFeedback` in the service parameters.
- Integrated `messageFeedback` into the `HoverButtons` component to conditionally render feedback buttons based on the interface configuration.
- Enhanced the loading of default interface settings to include `messageFeedback` support.

timmanik commented:
@rubentalstra good idea, I just updated the code to follow that design. Let me know your thoughts. Thanks!
